config/authorino/crd/patches/oneof_in_authconfigs.yaml

```diff
@@ -0,0 +1,57 @@
+# Enables oneOf validation for the identity/metadata/authorization modes
+
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/identity/items/oneOf
```

Comment:
only one identity can be specified for each authconfig object? authorino supports many at the same time.

Reply:
Not quite. It is actually defined in https://github.com/Kuadrant/authorino/blob/main/install/crd/patches/oneof_in_authconfigs.yaml. It means exactly one of the supported identity modes must be specified – either `apiKey`, or `oidc`, or `oauths`, or `kubernetes`.

Reply:
ok, I see. It is not about the `spec.identity` array, it is about the `items` of it. They must comply with one and only one of the specs there. 

The validation accepts empty array:
```yaml
---                                      
apiVersion: authorino.3scale.net/v1beta1 
kind: AuthConfig                         
metadata:                                
  name: my-api-protection                
spec:                                    
  hosts:                                 
  ⦙ - my-api.io                          
  identity: []                           

```

Reply:
Right now empty array means always 401. Hopefully #112 will fix that.